app/static/js/app.js

```diff
@@ -293,6 +293,21 @@ function setCursor(cursor, save = true) {
   }
 }
 
+function setFullScreen() {
+  let el = document.getElementById("remote-screen");
+  if (!el.dataset.onfullscreenchange) {
```

Comment:
Can we add a comment here? I'm not totally understanding what this is doing.

Reply:
Which lines exactly there is a lot happening here. Tell me how you would like to word it. 

`function setFullScreen() {
  const remoteScreen = document.getElementById("remote-screen");
  const remoteScreenImg = document.getElementById("remote-screen-img");
  //check to see if "onfullscreenchange" event has been created
  if (!remoteScreen.dataset.onfullscreenchange) {
    //if not continue and mark it as crated
    remoteScreen.dataset.onfullscreenchange = true;
    //create the "onfullscreenchange" event
    remoteScreen.onfullscreenchange = (evt) => {
      const remoteScreen = evt.target;
      const remoteScreenImg = document.getElementById("remote-screen-img");
      //if remoteScreen is still the fullscreenelement, then disregard
      if (document.fullscreenElement !== remoteScreen) {
        //in order for this to work properly I am polling the /state endpoint for the resolution of the screen
        //if we are not in full screen mode we do not need to poll anymore.
        clearImgSizeTimer();
        //disable fullscreen to toggle the class
        remoteScreen.setAttribute("fullscreen", false);
        //clear the custom styling
        remoteScreenImg.style.width = null;
        remoteScreenImg.style["max-width"] = null;
        remoteScreenImg.style["max-height"] = null;
      }
    };
  }
  remoteScreen.setAttribute("fullscreen", true);
  remoteScreenImg.style.width = streamState[1] + "px";
  remoteScreenImg.style["max-width"] = streamState[1] + "px";
  remoteScreenImg.style["max-height"] = streamState[2] + "px";
  remoteScreen.requestFullscreen();
}`